packages/api/src/command/medical/patient/update-patient.ts

```diff
@@ -22,42 +24,130 @@ export type PatientUpdateCmd = BaseUpdateCmdWithCustomer &
 // See: document-query.test.ts, "send a modified object to Sequelize"
 // See: https://metriport.slack.com/archives/C04DMKE9DME/p1686779391180389
-export async function updatePatient(
-  patientUpdate: PatientUpdateCmd,
+export async function updatePatient({
+  patientUpdate,
   emit = true,
+  rerunPdOnNewDemographics = false,
+  augmentDemographics = false,
+  forceCommonwell,
+  forceCarequality,
+}: {
+  patientUpdate: PatientUpdateCmd;
+  emit?: boolean;
+  rerunPdOnNewDemographics?: boolean;
+  augmentDemographics?: boolean;
   // START TODO #1572 - remove
-  forceCommonwell?: boolean,
-  forceCarequality?: boolean
+  forceCommonwell?: boolean;
+  forceCarequality?: boolean;
   // END TODO #1572 - remove
-): Promise<Patient> {
+}): Promise<Patient> {
   const { cxId, facilityId } = patientUpdate;
 
   // validate facility exists and cx has access to it
-  const facility = await getFacilityOrFail({ cxId, id: facilityId });
-
-  const requestId = uuidv7();
-
-  const patientUpdateWithPD: PatientUpdateCmd = {
-    ...patientUpdate,
-    patientDiscovery: { requestId, startedAt: new Date() },
-  };
+  await getFacilityOrFail({ cxId, id: facilityId });
 
-  const patient = await updatePatientWithoutHIEs(patientUpdateWithPD, emit);
+  const patient = await updatePatientWithoutHIEs(patientUpdate, emit);
 
   const fhirPatient = toFHIR(patient);
   await upsertPatientToFHIRServer(patientUpdate.cxId, fhirPatient);
 
-  await cqCommands.patient.discover(patient, facility.id, requestId, forceCarequality);
-
-  await cwCommands.patient.update(
+  const updatedPatient = await scheduleOrRunPatientDiscovery({
     patient,
     facilityId,
-    getCqOrgIdsToDenyOnCw,
-    requestId,
-    forceCommonwell
-  );
+    rerunPdOnNewDemographics,
+    augmentDemographics,
+    forceCommonwell,
+    forceCarequality,
+  });
+
+  return updatedPatient;
+}
+
+// Needs to live somewhere else
+export async function scheduleOrRunPatientDiscovery({
+  patient,
+  facilityId,
+  rerunPdOnNewDemographics,
+  augmentDemographics,
+  forceCommonwell,
+  forceCarequality,
+  isRerunFromNewDemographics = false,
+  overrideSchedule = false,
+}: {
+  patient: Patient;
+  facilityId: string;
+  rerunPdOnNewDemographics: boolean;
+  augmentDemographics: boolean;
+  // START TODO #1572 - remove
+  forceCommonwell?: boolean;
+  forceCarequality?: boolean;
+  // END TODO #1572 - remove
+  overrideSchedule?: boolean;
+  isRerunFromNewDemographics?: boolean;
+}): Promise<Patient> {
+  let updatedPatient = patient;
+  const cqData = cqCommands.patient.getData(patient.data.externalData);
+
+  const discoveryStatusCq = cqData?.discoveryStatus;
+  const scheduledPdRequestCq = cqData?.scheduledPdRequest;
+
+  if (
+    discoveryStatusCq === "processing" &&
+    (!scheduledPdRequestCq || (scheduledPdRequestCq && overrideSchedule))
+  ) {
+    updatedPatient = await schedulePatientDiscovery({
+      requestId: uuidv7(),
```

Comment:
ditto: use the same `requestId` across diff HIEs (here and below, for CW)

Reply:
Fixed. Note main thread.